src/Corax/Indexing/IndexWriter.cs

```diff
@@ -1122,8 +1124,14 @@ private void InsertSpatialField(Tree entriesToSpatialTree, IndexedField indexedF
             var termContainerId = fieldRootPage << 3 | 0b010;
             Debug.Assert(termContainerId >>> 3 == fieldRootPage, "field root too high?");
             var entriesToTerms = entriesToSpatialTree.FixedTreeFor(indexedField.Name, sizeof(double)+sizeof(double));
+
+
+            long processedTermsCounter = 0L;
             foreach (var (entry, spatialEntry)  in indexedField.Spatial)
             {
+                if ((++processedTermsCounter & Constants.IndexWriter.CancellationTokenCheckInterval) == 0L)
```

Comment:
I'm pretty sure that `ThrowIfCancellationRequested` will already inline this check - so there is no need to do this.